clientpool/channel.go

```diff
@@ -181,3 +220,24 @@ func (cp *channelPool) NumAllocated() int32 {
 func (cp *channelPool) IsExhausted() bool {
 	return cp.NumActiveClients() >= int32(cp.maxClients)
 }
+
+func (cp *channelPool) ensureMinClients() {
+	for !cp.isClosed.Load() {
+		time.Sleep(cp.backgroundTaskInterval)
```

Comment:
create a [`time.Ticker`](https://pkg.go.dev/time#Ticker) to loop through instead of using sleep, you can use closed signal as the condition of termination of the loop.